src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -312,6 +309,9 @@ tuple<set<Cell>, set<Cell>, vector<Segment>> TemporalMemory::burstColumns(
     _unpredictedColumns.resize(it - _unpredictedColumns.begin());
   }
 
+  // Sort unpredictedActiveColumns before iterating for python compatibility
+  sort(_unpredictedColumns.begin(), _unpredictedColumns.end());
```

Comment:
The Python version of this has a secondary key. Do we need to add that here as well?